pkg/resourcemanager/rediscaches/actions/rediscacheaction_reconcile.go

```diff
@@ -34,40 +34,49 @@ func (m *AzureRedisCacheActionManager) Ensure(ctx context.Context, obj runtime.O
 		return true, nil
 	}
 
-	rollAllKeys := instance.Spec.ActionName == v1alpha1.RedisCacheActionNameRollAllKeys
+	if v1alpha1.IsRedisCacheRollAction(instance.Spec.ActionName) {
+		rollAllKeys := instance.Spec.ActionName == v1alpha1.RedisCacheActionNameRollAllKeys
+
+		if rollAllKeys || instance.Spec.ActionName == v1alpha1.RedisCacheActionNameRollPrimaryKey {
+			if err = m.RegeneratePrimaryAccessKey(ctx, instance.Spec.ResourceGroup, instance.Spec.CacheName); err != nil {
+				instance.Status.Message = err.Error()
+				return false, err
+			}
+		}
+
+		if rollAllKeys || instance.Spec.ActionName == v1alpha1.RedisCacheActionNameRollSecondaryKey {
+			if err = m.RegenerateSecondaryAccessKey(ctx, instance.Spec.ResourceGroup, instance.Spec.CacheName); err != nil {
+				instance.Status.Message = err.Error()
+				return false, err
+			}
+		}
 
-	if rollAllKeys || instance.Spec.ActionName == v1alpha1.RedisCacheActionNameRollPrimaryKey {
-		if err = m.RegeneratePrimaryAccessKey(ctx, instance.Spec.ResourceGroup, instance.Spec.CacheName); err != nil {
+		cacheInstance := &v1alpha1.RedisCache{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      instance.Spec.CacheName,
+				Namespace: instance.Namespace,
+			},
+			Spec: v1alpha1.RedisCacheSpec{
+				SecretName:        instance.Spec.SecretName,
+				ResourceGroupName: instance.Spec.ResourceGroup,
+			},
+		}
+		if err = m.ListKeysAndCreateSecrets(ctx, cacheInstance); err != nil {
+			instance.Status.Provisioned = false
```

Comment:
make sure that provisioning cant be true here... if it can, set to false

Reply:
I intentionally didn't set provisioning because all of the actions thus far have been synchronous. Should I make sure it's set to false anyways?